### PR TITLE
aboot: fastboot-extra: Add "fastboot oem readl&writel&readb&writeb"

### PR DESCRIPTION
**Pull request by fxsheep**
Add fastboot oem commands to read/write a specified address.

Usage:

```
fastboot oem readl <address>
fastboot oem readb <address>
fastboot oem writel <address> <value>
fastboot oem writeb <address> <value>
```

### CODE REVIEW

File: app/aboot/fastboot-extra.c [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#include <stdlib.h>
```
Comment (stephan-gh) [inline app/aboot/fastboot-extra.c:1]: The include list is alphabetically ordered

File: app/aboot/fastboot-extra.c [hunk 2]
```diff
@@ -13,6 +14,87 @@
...
+		}
+		if (arg[i] == ' ' && i <= 10 && address[0] == 0) {
+			strncpy(address, arg, i);
+			addr = atoi(address);
+		}
+		if (arg[i] == ' ' && arg[i + 1] != ' ') {
+			value = atoi(arg + i + 1);
+			break;
+		}
+	}
```
Comment (stephan-gh) [inline app/aboot/fastboot-extra.c:48]: Looks like we can share this code between writel and writeb. I'm also a bit confused about how it works... Perhaps we should simply use the standard `strtok` here? Something like this:

```c
static bool parse_write_args(const char *arg, uint32_t *addr, uint32_t *value)
{
	char *saveptr;
	char *args = strdup(arg);
	char *addr_str = strtok_r(args, " ", &saveptr);
	char *value_str = strtok_r(NULL, " ", &saveptr);

	if (!addr_str || !value_str) {
		free(args);
		return false;
	}

	*addr = atoi(addr_str);
	*value = atoi(value_str);
	free(args);
	return true;
}
```

**Review by stephan-gh — COMMENTED** [review]
Thanks for the PR! Just two small comments.

File: app/aboot/fastboot-extra.c (hunk 2 above)
Comment (fxsheep) [reply app/aboot/fastboot-extra.c:48]: > we can share this code between writel and writeb. 

Yeah, I thought about that too. 

>we should simply use the standard strtok here

Ah, I thought that won't work with more than one space. Turned out I thought too much.
Thanks for the review!

**Review by stephan-gh — APPROVED** [review]

**Review by TravMurav — APPROVED** [review]

**Pull request merged**